app/Http/Controllers/Auth/LoginController.php

```diff
@@ -50,17 +50,38 @@ public function __construct()
         \Session::put('backUrl', \URL::previous());
     }
 
-
     function showLoginForm()
     {
+        $this->loginViaRemoteUser();
         if (Auth::check()) {
             return redirect()->intended('dashboard');
         }
+
+        if (Setting::getSettings()->login_common_disabled == "1") {
+            return view('errors.401');
+        }
+
         return view('auth.login');
     }
 
+    private function loginViaRemoteUser()
+    {
+        if (Setting::getSettings()->login_remote_user_enabled == "1"
+            && isset($_SERVER['REMOTE_USER']) && !empty($_SERVER['REMOTE_USER'])) {
```

Comment:
Codacy is complaining about the use of $_SERVER.  It looks like laravel exposes a Request::server() method that may be more appropriate. https://laravel.com/api/5.4/Illuminate/Http/Request.html#method_server

Not sure it's a terribly huge deal.

Reply:
It's not, but I prefer to use the Laravel-isms when possible.